lib/modules/manager/puppet/puppetfile-parser.ts

```diff
@@ -0,0 +1,79 @@
+import { regEx } from '../../../util/regex';
+import type { PuppetForgeUrl, Puppetfile, PuppetfileModule } from './types';
+
+const lineTerminationRegex = regEx(`\r?\n`);
+const forgeRegex = regEx(/^forge\s+['"]([^'"]+)['"]/);
+
+export function parsePuppetfile(content: string): Puppetfile {
+  const puppetfile: Puppetfile = new Map<PuppetForgeUrl, PuppetfileModule[]>();
+
+  let currentForge = undefined;
```

Comment:
```suggestion
  let currentForge: string | undefined = undefined;
```

Reply:
done

Reply:
> done

Please read 👉 https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting